File: rendezvous-service/src/main/java/com/medical/entity/RendezVous.java
package com.medical.entity;

import jakarta.persistence.*;
import lombok.*;
import com.medical.enums.StatusRendezVous;
import com.medical.enums.TypeRendezVous;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "rendez_vous")
public class RendezVous {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idRendezVous;

    private String dateRendezVous;
    private String heureRdv;

    private Long idPatient;
    private Long idMedecin;

    @Enumerated(EnumType.STRING)
    private StatusRendezVous status;

    private String notes;
    @Enumerated(EnumType.STRING)
    private TypeRendezVous typeRendezVous;
}

File: frontend/src/types/rendezvous.ts
export interface RendezVous {
    idRendezVous?: number;
    dateRendezVous: string;
    heureRdv: string;
    idPatient: number;
    idMedecin: number;
    status: 'PLANIFIE' | 'CONFIRME' | 'ANNULE' | 'TERMINE';
    motif?: string;
    notes?: string;
}

export interface Patient {
    idPatient: number;
    nom: string;
    prenom: string;
    email: string;
    telephone?: string;
}

export interface Medecin {
    idMedecin: number;
    nom: string;
    prenom?: string;
    specialite: string;
    email?: string;
}

export type StatusRendezVous = 'PLANIFIE' | 'CONFIRME' | 'ANNULE' | 'TERMINE';